modules/dbnd/src/targets/caching.py

```diff
@@ -226,3 +218,25 @@ def get_or_create(file_target):
             dbnd_meta_cache = DbndLocalFileMetadataRegistry(file_path=file_target.path)
             dbnd_meta_cache.save()
             return dbnd_meta_cache
+
+    @staticmethod
+    def refresh(file_target):
+        dbnd_meta_cache = DbndLocalFileMetadataRegistry.get_or_create(file_target)
+        if dbnd_meta_cache.expired:
+            dbnd_meta_cache.delete()
+            return DbndLocalFileMetadataRegistry.get_or_create(file_target)
+
+
+def get_or_create_folder_in_dbnd_local_root(folder_name):
+    # type: (str) -> str
+
+    run = try_get_databand_run()
+    if not run:
+        raise Exception("No databand run found to when creating cache file")
+    dbnd_local_root = run.get_current_dbnd_local_root()
```

Comment:
Too much work for getting the local root, how about:
`task.task_env.dbnd_local_root` or in this class
`self.task_run.task.task_env.dbnd_local_root`

This way you can avoid calling `try_get_databand_run` and handling None case, etc.

Reply:
It seems like a lot of work to get to root but on the other hand it simplifies `DbndLocalFileMetadataRegistry` class. Keeping task/dbnd_local_root info in this class would be troublesome as it has a lot of static and class methods.

I think its best not to change too much in this class but I can change `get_or_create_folder_in_dbnd_local_root` so it takes `dbnd_local_root` as an argument so we do not have to change `DbndLocalFileMetadataRegistry` too much to preserve its behaviour and we do not have to extract `databand_run` in `get_or_create_folder_in_dbnd_local_root`